--- upp/configure	Wed Dec 18 16:28:42 2024
+++ upp-patched/configure	Mon Feb 24 13:27:22 2025
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd ${SCRIPT_DIR}
+
 ./configure_makefile Makefile
 ./configure_makefile umkMakefile
 
--- upp/Makefile.in	Wed Dec 18 16:28:58 2024
+++ upp-patched/Makefile.in	Wed Feb 26 17:17:49 2025
@@ -2,16 +2,20 @@
 UPPDIR2 = /home/cxl/.config/u++/umk/UppHub/gdal/
 UPPDIR3 = /home/cxl/.config/u++/umk/UppHub/eigen/
 
+OI_COMMON = -m64
+OI_INC = -I/usr/gcc/14/include/c++/14.2.0/ -I/usr/gcc/14/include/c++/14.2.0/x86_64-pc-solaris2.11 -I/usr/clang/18/include
+OI_LDFLAGS = -Wl,-rpath=/usr/lib -Wl,-rpath=/usr/clang/18/lib -L/usr/gcc/14/lib/gcc/x86_64-pc-solaris2.11/14.2.0/ -L/usr/clang/18/lib
+
 UPPOUT = .cache/upp.out/
-CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` -I./ -I$(UPPOUT)
-Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
+CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` $(OI_INC) -I./ -I$(UPPOUT)
+Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagSOLARIS -DflagLCLANG -DflagUSEMALLOC
+CXX = /usr/gcc/14/bin/g++
 LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = $(LINKOPTIONS)
+CFLAGS = -O3 -ffunction-sections -fdata-sections $(OI_COMMON)
+CXXFLAGS = -O3 -ffunction-sections -fdata-sections -std=c++17 $(OI_COMMON)
+LDFLAGS = $(OI_COMMON) $(OI_LDFLAGS) $(LINKOPTIONS)
 LIBPATH =
-AR = ar -sr
+AR = /usr/gcc/14/bin/gcc-ar -sr
 
 OutDir_ide = $(UPPOUT)ide/GCC-Gcc-Gui-Linux-Main-Posix-Shared/
 Macro_ide = $(Macro) -DflagMAIN
@@ -400,6 +404,7 @@
 			-lpthread \
 			-ldl \
 			-lrt \
+			-lclang \
 			-lz -Wl,--end-group
 
 $(OutDir_ide):
--- upp/Makefile	Wed Dec 18 16:28:58 2024
+++ upp-patched/Makefile	Wed Feb 26 17:17:54 2025
@@ -2,92 +2,96 @@
 UPPDIR2 = /home/cxl/.config/u++/umk/UppHub/gdal/
 UPPDIR3 = /home/cxl/.config/u++/umk/UppHub/eigen/
 
+OI_COMMON = -m64
+OI_INC = -I/usr/gcc/14/include/c++/14.2.0/ -I/usr/gcc/14/include/c++/14.2.0/x86_64-pc-solaris2.11 -I/usr/clang/18/include
+OI_LDFLAGS = -Wl,-rpath=/usr/lib -Wl,-rpath=/usr/clang/18/lib -L/usr/gcc/14/lib/gcc/x86_64-pc-solaris2.11/14.2.0/ -L/usr/clang/18/lib
+
 UPPOUT = .cache/upp.out/
-CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` -I./ -I$(UPPOUT)
-Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
+CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` $(OI_INC) -I./ -I$(UPPOUT)
+Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagSOLARIS -DflagLCLANG -DflagUSEMALLOC
+CXX = /usr/gcc/14/bin/g++
 LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = -Wl,--gc-sections $(LINKOPTIONS)
+CFLAGS = -O3 -ffunction-sections -fdata-sections $(OI_COMMON)
+CXXFLAGS = -O3 -ffunction-sections -fdata-sections -std=c++17 $(OI_COMMON)
+LDFLAGS = $(OI_COMMON) $(OI_LDFLAGS) $(LINKOPTIONS)
 LIBPATH =
-AR = ar -sr
+AR = /usr/gcc/14/bin/gcc-ar -sr
 
-OutDir_ide = $(UPPOUT)ide/GCC-Gcc-Gui-Linux-Main-Posix-Shared/
+OutDir_ide = $(UPPOUT)ide/GCC-Gcc-Gui-Main-Posix-Shared-Solaris/
 Macro_ide = $(Macro) -DflagMAIN
-OutDir_ide_Common = $(UPPOUT)ide/Common/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Common = $(UPPOUT)ide/Common/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Common = $(Macro)
-OutDir_ide_Core = $(UPPOUT)ide/Core/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Core = $(UPPOUT)ide/Core/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Core = $(Macro)
-OutDir_ide_LayDes = $(UPPOUT)ide/LayDes/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_LayDes = $(UPPOUT)ide/LayDes/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_LayDes = $(Macro)
-OutDir_ide_Builders = $(UPPOUT)ide/Builders/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Builders = $(UPPOUT)ide/Builders/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Builders = $(Macro)
-OutDir_ide_Debuggers = $(UPPOUT)ide/Debuggers/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Debuggers = $(UPPOUT)ide/Debuggers/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Debuggers = $(Macro)
-OutDir_ide_Browser = $(UPPOUT)ide/Browser/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Browser = $(UPPOUT)ide/Browser/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Browser = $(Macro)
-OutDir_CodeEditor = $(UPPOUT)CodeEditor/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_CodeEditor = $(UPPOUT)CodeEditor/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_CodeEditor = $(Macro)
-OutDir_CtrlLib = $(UPPOUT)CtrlLib/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_CtrlLib = $(UPPOUT)CtrlLib/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_CtrlLib = $(Macro)
-OutDir_plugin_bz2 = $(UPPOUT)plugin/bz2/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_bz2 = $(UPPOUT)plugin/bz2/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_bz2 = $(Macro)
-OutDir_HexView = $(UPPOUT)HexView/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_HexView = $(UPPOUT)HexView/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_HexView = $(Macro)
-OutDir_TextDiffCtrl = $(UPPOUT)TextDiffCtrl/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_TextDiffCtrl = $(UPPOUT)TextDiffCtrl/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_TextDiffCtrl = $(Macro)
-OutDir_TabBar = $(UPPOUT)TabBar/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_TabBar = $(UPPOUT)TabBar/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_TabBar = $(Macro)
-OutDir_ide_Designers = $(UPPOUT)ide/Designers/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Designers = $(UPPOUT)ide/Designers/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Designers = $(Macro)
-OutDir_ide_Android = $(UPPOUT)ide/Android/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Android = $(UPPOUT)ide/Android/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Android = $(Macro)
-OutDir_ide_Java = $(UPPOUT)ide/Java/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_Java = $(UPPOUT)ide/Java/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_Java = $(Macro)
-OutDir_ide_MacroManager = $(UPPOUT)ide/MacroManager/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_MacroManager = $(UPPOUT)ide/MacroManager/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_MacroManager = $(Macro)
-OutDir_Report = $(UPPOUT)Report/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Report = $(UPPOUT)Report/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Report = $(Macro)
-OutDir_Core_SSL = $(UPPOUT)Core/SSL/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Core_SSL = $(UPPOUT)Core/SSL/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Core_SSL = $(Macro)
-OutDir_plugin_md = $(UPPOUT)plugin/md/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_md = $(UPPOUT)plugin/md/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_md = $(Macro)
-OutDir_ide_clang = $(UPPOUT)ide/clang/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_clang = $(UPPOUT)ide/clang/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_clang = $(Macro)
-OutDir_Esc = $(UPPOUT)Esc/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Esc = $(UPPOUT)Esc/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Esc = $(Macro)
-OutDir_Core = $(UPPOUT)Core/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Core = $(UPPOUT)Core/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Core = $(Macro)
-OutDir_plugin_lzma = $(UPPOUT)plugin/lzma/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_lzma = $(UPPOUT)plugin/lzma/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_lzma = $(Macro)
-OutDir_plugin_lz4 = $(UPPOUT)plugin/lz4/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_lz4 = $(UPPOUT)plugin/lz4/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_lz4 = $(Macro)
-OutDir_plugin_zstd = $(UPPOUT)plugin/zstd/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_zstd = $(UPPOUT)plugin/zstd/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_zstd = $(Macro)
-OutDir_RichEdit = $(UPPOUT)RichEdit/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_RichEdit = $(UPPOUT)RichEdit/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_RichEdit = $(Macro)
-OutDir_plugin_png = $(UPPOUT)plugin/png/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_png = $(UPPOUT)plugin/png/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_png = $(Macro)
-OutDir_Draw = $(UPPOUT)Draw/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Draw = $(UPPOUT)Draw/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Draw = $(Macro)
-OutDir_PdfDraw = $(UPPOUT)PdfDraw/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_PdfDraw = $(UPPOUT)PdfDraw/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_PdfDraw = $(Macro)
-OutDir_Painter = $(UPPOUT)Painter/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_Painter = $(UPPOUT)Painter/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_Painter = $(Macro)
-OutDir_plugin_pcre = $(UPPOUT)plugin/pcre/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_pcre = $(UPPOUT)plugin/pcre/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_pcre = $(Macro)
-OutDir_CtrlCore = $(UPPOUT)CtrlCore/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_CtrlCore = $(UPPOUT)CtrlCore/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_CtrlCore = $(Macro)
-OutDir_plugin_bmp = $(UPPOUT)plugin/bmp/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_bmp = $(UPPOUT)plugin/bmp/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_bmp = $(Macro)
-OutDir_RichText = $(UPPOUT)RichText/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_RichText = $(UPPOUT)RichText/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_RichText = $(Macro)
-OutDir_ide_IconDes = $(UPPOUT)ide/IconDes/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_ide_IconDes = $(UPPOUT)ide/IconDes/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_ide_IconDes = $(Macro)
-OutDir_plugin_gif = $(UPPOUT)plugin/gif/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_gif = $(UPPOUT)plugin/gif/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_gif = $(Macro)
-OutDir_plugin_jpg = $(UPPOUT)plugin/jpg/GCC-Gcc-Gui-Linux-Posix-Shared/
+OutDir_plugin_jpg = $(UPPOUT)plugin/jpg/GCC-Gcc-Gui-Posix-Shared-Solaris/
 Macro_plugin_jpg = $(Macro)
 
 OutDir = $(OutDir_ide)
@@ -262,7 +266,7 @@
 	$(OutDir_plugin_gif)gif.a \
 	$(OutDir_plugin_jpg)jpgreg.o \
 	$(OutDir_plugin_jpg)jpg.a
-	$(LINKER) -o "$(OutFile)" -Wl,-s $(LIBPATH) -Wl,-O,2 $(LDFLAGS) -Wl,--start-group  \
+	$(LINKER) -o "$(OutFile)" -Wl,-s $(LIBPATH) $(LDFLAGS) -Wl,--start-group  \
 		$(OutDir_ide)BaseDlg.o \
 		$(OutDir_ide)SelectPkg.o \
 		$(OutDir_ide)UppWspc.o \
@@ -397,9 +401,10 @@
 			-lbz2 \
 			-lcrypto \
 			-lssl \
-			-lpthread \
+			-lposix4 \
 			-ldl \
-			-lrt \
+			-lsocket -lnsl \
+			-lclang \
 			-lz -Wl,--end-group
 
 $(OutDir_ide):
